feature-gate/program/src/instruction.rs

```diff
@@ -0,0 +1,87 @@
+//! Program instructions
+
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+    },
+};
+
+/// Feature Gate program instructions
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
```

Comment:
Can we use `num_enum` for this instead of `borsh`? It'll be much easier for encoding a u8